src/components/Avatar/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { useAppDispatch } from 'store/hooks';
+import { changeUserAvatar } from 'controllers/user';
+import {
+    Image, Modal, Form, Button,
+} from 'react-bootstrap';
+
+interface IAvatarProps {
+    src: string;
+}
+
+export const Avatar = ({ src }: IAvatarProps) => {
+    const dispatch = useAppDispatch();
+    const [modalShow, setModalShow] = useState(false);
+    const [file, setFile] = useState(null);
+
+    const onFileChange = (event: any) => {
+        setFile(event.target.files[0]);
+    };
```

Comment:
Вроде уже у писал, что в TS нет причин использовать `any`. Оно отключает проверку типов в месте использования.
Для аргумента можно использовать тип `Event`, а `target` кастануть к `HTMLInputElement`, чтобы не ругалсян на отсутствие `files`


```suggestion
    const onFileChange = (event: Event) => {
        const target = event.target as HTMLInputElement
        setFile(target.files[0]);
    };
```